src/Compilers/Core/CodeAnalysisTest/ObjectSerializationTests.cs

```diff
@@ -500,6 +500,36 @@ public void TestPrimitiveArrayValues()
             TestRoundTrip(w => TestWritingPrimitiveArrays(w), r => TestReadingPrimitiveArrays(r));
         }
 
+        [Fact]
+        public void TestEmptyByteSpan()
+        {
+            TestRoundTrip(w => TestWritingByteSpan(Array.Empty<byte>(), w), r => TestReadingByteSpan(Array.Empty<byte>(), r));
+        }
+
+        [Fact]
+        public void TestByteSpan()
```

Comment:
One more thought: can we test the three special cases as well (arity 1, 2 and 3)? Thanks

Reply:
Added the tests, and consolidated down to a single `[Theory]` with specified combinatorial values.